packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/layout.ts

```diff
@@ -29,67 +29,77 @@ export const performLayout = (
+  nodes
+    .filter((node) => !node.hidden)
+    .forEach((node) => {
+      const elkNode: ElkNode = {
+        id: node.id,
+        children: [],
+        labels: [],
+        layoutOptions: {
+          'nodeLabels.placement': '[H_CENTER, V_TOP, INSIDE]',
+        },
       };
 
-      elkNode.labels.push(elkLabel);
-    }
+      if (node.type === 'rectangularNode') {
+        const rectangularNodeData: RectangularNodeData = node.data as RectangularNodeData;
 
-    const element = document.querySelector(`[data-id="${node.id}"]`);
-    if (element) {
-      elkNode.width = element.clientWidth;
-      elkNode.height = element.clientHeight;
-    }
+        const label = document.querySelector<HTMLDivElement>(`[data-id="${rectangularNodeData.label.id}"]`);
+        const elkLabel: ElkLabel = {
+          width: label.getBoundingClientRect().width,
+          height: label.getBoundingClientRect().height,
+          text: rectangularNodeData.label.text,
+        };
 
-    nodeId2ElkNode.set(elkNode.id, elkNode);
-  });
+        elkNode.labels.push(elkLabel);
+      }
 
-  nodes.forEach((node) => {
-    if (graph.children) {
-      if (!!node.parentNode) {
-        const elknodeChild = nodeId2ElkNode.get(node.id);
-        const elkNodeParent = nodeId2ElkNode.get(node.parentNode);
-        elkNodeParent.children.push(elknodeChild);
-      } else {
-        const elkNodeRoot = nodeId2ElkNode.get(node.id);
-        graph.children.push(elkNodeRoot);
+      const element = document.querySelector(`[data-id="${node.id}"]`);
+      if (element) {
+        elkNode.width = element.clientWidth;
+        elkNode.height = element.clientHeight;
       }
-    }
-  });
 
-  edges.forEach((edge) => {
-    if (graph.edges) {
-      const elkEdge: ElkExtendedEdge = {
-        id: edge.id,
-        sources: [edge.source],
-        targets: [edge.target],
-      };
-      graph.edges.push(elkEdge);
-    }
-  });
+      nodeId2ElkNode.set(elkNode.id, elkNode);
+    });
+
+  nodes
+    .filter((node) => !node.hidden)
+    .forEach((node) => {
+      if (graph.children) {
+        if (!!node.parentNode) {
+          const elknodeChild = nodeId2ElkNode.get(node.id);
+          const elkNodeParent = nodeId2ElkNode.get(node.parentNode);
+          if (elkNodeParent && elkNodeParent.children) {
+            elkNodeParent.children.push(elknodeChild);
+          }
+        } else {
+          const elkNodeRoot = nodeId2ElkNode.get(node.id);
+          graph.children.push(elkNodeRoot);
+        }
+      }
+    });
+
+  edges
+    .filter((edge) => !edge.hidden)
+    .forEach((edge) => {
+      if (graph.edges) {
+        const elkEdge: ElkExtendedEdge = {
+          id: edge.id,
+          sources: [edge.source],
+          targets: [edge.target],
+        };
+        graph.edges.push(elkEdge);
+      }
+    });
 
   return elk.layout(graph).then((layoutedGraph) => {
-    const nodes: Node[] = [];
+    const nodes: Node[] = [...hiddenNodes];
```

Comment:
I do not like to initialize nodes with hidden nodes that way.
I prefer to leverage on promise API and use a `then` to add hidden nodes after elk nodes have been transformed into react flow nodes.